webapp/src/test/java/info/rmapproject/webapp/controllers/SitePropertiesToModelInterceptorTest.java

```diff
@@ -0,0 +1,130 @@
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.springframework.web.servlet.ModelAndView;
+
+import info.rmapproject.webapp.WebTestAbstract;
+import info.rmapproject.webapp.utils.SiteProperties;
+
+/**
+ * Tests for SessionDefaultsInterceptor
+ * @author khanson
+ *
+ */
+
+public class SitePropertiesToModelInterceptorTest extends WebTestAbstract {
+
+	@Autowired
+    private WebApplicationContext wac;
+	
+	private MockMvc mockMvc;
+	
+	@Before
+	public void setup() {
+    	mockMvc = MockMvcBuilders.webAppContextSetup(wac).build();
+	}
+		
+	/**
+	 * Simple test that uses the constructor to set the attribute values, then checks they have been added to the request session attributes
+	 */
+	@Test
+	public void testDefaultsAppliedThroughConstructor() {
+	    try {
+	        MockHttpServletRequest request = new MockHttpServletRequest("GET","/home");
+	        MockHttpServletResponse response = new MockHttpServletResponse();	
+	        ModelAndView modelview = new ModelAndView();
+	        SiteProperties siteProperties = new SiteProperties(true, false, false);
+	        SitePropertiesToModelInterceptor interceptor = new SitePropertiesToModelInterceptor(siteProperties);
+	        interceptor.postHandle(request, response, null, modelview);
+	        SiteProperties siteProps = (SiteProperties) modelview.getModel().get(SITE_PROPERTIES_ATTRIBNAME);
+	        assertTrue(siteProps.isGoogleEnabled());
+	        assertFalse(siteProps.isTwitterEnabled());
+	        assertFalse(siteProps.isOrcidEnabled());
+	        	        
+	        request = new MockHttpServletRequest("GET","/home");
+	        response = new MockHttpServletResponse();
+	        siteProperties = new SiteProperties(false, true, false);
+	        interceptor = new SitePropertiesToModelInterceptor(siteProperties);
+	        interceptor.postHandle(request, response, null, modelview);
+	        siteProps = (SiteProperties) modelview.getModel().get(SITE_PROPERTIES_ATTRIBNAME);
+	        assertFalse(siteProps.isGoogleEnabled());
+	        assertTrue(siteProps.isOrcidEnabled());
+	        assertFalse(siteProps.isTwitterEnabled());
+
+	        request = new MockHttpServletRequest("GET","/home");
+	        response = new MockHttpServletResponse();
+	        siteProperties = new SiteProperties(false, false, true);
+	        interceptor = new SitePropertiesToModelInterceptor(siteProperties);
+	        interceptor.postHandle(request, response, null, modelview);
+	        siteProps = (SiteProperties) modelview.getModel().get(SITE_PROPERTIES_ATTRIBNAME);
+	        assertFalse(siteProps.isGoogleEnabled());
+	        assertFalse(siteProps.isOrcidEnabled());
+	        assertTrue(siteProps.isTwitterEnabled());
+
+	        	        		        
+	    } catch (Exception e) {

```

Comment:
So these catch blocks aren't really necessary.

If something in the `try` block throws an exception, the exception can simply be declared in the method signature using a `throws` statement, and JUnit will handle the rest by failing the test and emitting the stacktrace (which will be present in the surefire report).  

So printing the stack trace and then calling fail() emulates the existing behavior of JUnit and would be considered redundant.  Removing the try/catch and the redundant exception handling makes the test a bit more readable!